### ISSUE TITLE
Kubernetes 1.11 problems

### ISSUE THREAD

**Issue by schmitch** [zalando-incubator/postgres-operator#339]
Hello, I tried to use postgres-operator on a 1.11 cluster.

However bootstrapping fails with:

```
Could not take out TTL lock
2018-07-08 12:30:39,465 INFO: following new leader after trying and failing to obtain lock
```

I tried it on two 1.11 clusters, however one cluster already has a running patroni over endpoints (which still works fine), however it got started on k8s 1.10.

currently I can bootstrap postgres-operator on k8s 1.10 (minikube) without problems (with the same manifests)

**Comment by alexeyklyukin** [comment]
Which version of Spilo are you running? There used to be an issue with Kubernetes 1.10 with endpoints during the bootstrap 
(https://github.com/zalando/patroni/commit/3eeb4ed97922ff1535ffabc49d5a7c35248bfdf8). The fix has been included in Patroni 1.4.4 and registry.opensource.zalan.do/acid/spilo-10.1.4-p8 (although you should try with the latest spilo, which is registry.opensource.zalan.do/acid/spilo-10.1.4-p11 anyway). 

Are you running 1.11 over endpoints as well? Is it OpenShift by any chance?

**Comment by schmitch** [comment]
ah no I just actually deleted some stuff by accident from the config.
By copy the configs into a git directory:
i.e. if:
```
 cluster_labels: application:spilo
  cluster_name_label: version
  pod_role_label: spilo-role
```
is missing there will be `Could not take out TTL lock` errors.
